plugins/trivy/pkg/image/getimages.go

```diff
@@ -36,100 +34,71 @@ func namespaceIsBlocked(ns string, namespaceBlocklist, namespaceAllowlist []stri
+		for _, p := range controller.Pods {
+			podUnstructuredContent := p.UnstructuredContent()
+			var pod corev1.Pod
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(podUnstructuredContent, &pod)
 			if err != nil {
-				logrus.Warnf("Error retrieving mapping %s of API %s and Kind %s because of error: %v ", firstOwner.Name, firstOwner.APIVersion, firstOwner.Kind, err)
-				break
+				logrus.Warnf("Unable to retrieve structured pod data: %v", err)
 			}
-			getParents, err := dynamicClient.Resource(mapping.Resource).Namespace(pod.ObjectMeta.Namespace).Get(ctx, firstOwner.Name, metav1.GetOptions{})
-			if err != nil {
-				logrus.Warnf("Error retrieving parent object %s of API %s and Kind %s because of error: %v ", firstOwner.Name, firstOwner.APIVersion, firstOwner.Kind, err)
-				break
-			}
-			owners = getParents.GetOwnerReferences()
-		}
 
-		for _, containerStatus := range pod.Status.ContainerStatuses {
-			var imageName string
-			if strings.HasPrefix(containerStatus.Image, "sha256") {
-				imageName = strings.TrimPrefix(containerStatus.ImageID, "docker-pullable://")
-				logrus.Debugf("using an image name %q from the containerStatuses.*.ImageID field, because containerStatuses.*.Image begins with sha256 - %q", imageName, containerStatus.Image)
-			} else {
-				imageName = containerStatus.Image
-			}
-			imageID := strings.TrimPrefix(containerStatus.ImageID, "docker-pullable://")
-			imagePullRef := imageID
-			if imagePullRef == "" || strings.HasPrefix(imagePullRef, "sha256:") {
-				imagePullRef = imageName
-			}
-			owner.Container = containerStatus.Name
-			imgKey := imageName + "/" + imageID
-			if imageOwners[imgKey] == nil {
-				imageOwners[imgKey] = map[models.Resource]struct{}{}
-			}
-			imageOwners[imgKey][owner] = struct{}{}
-			if _, found := keyToImage[imgKey]; found {
-				continue
-			}
-			keyToImage[imgKey] = models.Image{
-				ID:      imageID,
-				Name:    imageName,
-				PullRef: imagePullRef,
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				var imageName string
+				if strings.HasPrefix(containerStatus.Image, "sha256") {
+					imageName = strings.TrimPrefix(containerStatus.ImageID, "docker-pullable://")
+					logrus.Debugf("using an image name %q from the containerStatuses.*.ImageID field, because containerStatuses.*.Image begins with sha256 - %q", imageName, containerStatus.Image)
+				} else {
+					imageName = containerStatus.Image
+				}
+				imageID := strings.TrimPrefix(containerStatus.ImageID, "docker-pullable://")
+				imagePullRef := imageID
+				if imagePullRef == "" || strings.HasPrefix(imagePullRef, "sha256:") {
+					imagePullRef = imageName
+				}
+				owner.Container = containerStatus.Name
+				imgKey := imageName + "/" + imageID
+				if imageOwners[imgKey] == nil {
+					imageOwners[imgKey] = map[models.Resource]struct{}{}
+				}
+				imageOwners[imgKey][owner] = struct{}{}
+				if _, found := keyToImage[imgKey]; found {
+					continue
+				}
```

Comment:
Some of this mapping logic seems like it could probably be simplified now that we've changed the loop.

Another gotcha here is that, now that we're using controller-utils, not every Controller here will have pods to loop through. E.g. we may find a CronJob with no running pods. So we need a fallback image ID/name.

I think this is maybe worth splitting out into a follow-up though--I tried writing out the logic and it's not a trivial change. And the current PR is a strict improvement over how we're doing things now.

Reply:
Is doing the fallback logic (e.g. if a CronJob doesn't have an active pod scheduled) worth doing in this PR? Or is that also a bear along with the mapping logic for `imageOwners`?